Examples/Run/Common/src/CommonOptions.cpp

```diff
@@ -89,17 +89,30 @@ void ActsExamples::Options::addMaterialOptions(
 }
 
 void ActsExamples::Options::addOutputOptions(
-    boost::program_options::options_description& opt) {
+    boost::program_options::options_description& opt, int formats) {
   // Add specific options for this example
   opt.add_options()("output-dir", value<std::string>()->default_value(""),
-                    "Output directory location.")(
-      "output-root", bool_switch(),
-      "Switch on to write '.root' output file(s).")(
-      "output-csv", bool_switch(), "Switch on to write '.csv' output file(s).")(
-      "output-obj", bool_switch(), "Switch on to write '.obj' ouput file(s).")(
-      "output-json", bool_switch(),
-      "Switch on to write '.json' ouput file(s).")(
-      "output-txt", bool_switch(), "Switch on to write '.txt' ouput file(s).");
+                    "Output directory location.");
+
+  if ((OutputFormat::Root & formats) == OutputFormat::Root)
+    opt.add_options()("output-root", bool_switch(),
+                      "Switch on to write '.root' output file(s).");
+
+  if ((OutputFormat::Csv & formats) == OutputFormat::Csv)
+    opt.add_options()("output-csv", bool_switch(),
+                      "Switch on to write '.csv' output file(s).");
+
+  if ((OutputFormat::Obj & formats) == OutputFormat::Obj)
+    opt.add_options()("output-obj", bool_switch(),
+                      "Switch on to write '.obj' ouput file(s).");
+
+  if ((OutputFormat::Json & formats) == OutputFormat::Json)
+    opt.add_options()("output-json", bool_switch(),
+                      "Switch on to write '.json' ouput file(s).");
+
+  if ((OutputFormat::Txt & formats) == OutputFormat::Txt)
```

Comment:
See also [this](https://github.com/acts-project/acts/blob/master/Core/include/Acts/Utilities/Helpers.hpp#L27). Does exactly the same, but is a little bit more readable in my opinion.